File: back-end/src/main/java/com/huyphan/dtos/PostDto.java
package com.huyphan.dtos;

import com.huyphan.models.converters.PostContentTypeConverter;
import com.huyphan.models.enums.PostContentType;
import javax.persistence.Column;
import javax.persistence.Convert;
import javax.persistence.Lob;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@Getter
@Setter
public class PostDto {

    private Long id;

    private String title;

    private String mediaUrl;

    private String mediaType;
    private String text;

    private PostContentType contentType;

    private int downvotes;

    private int upvotes;

    private SectionDto section;

    private UserDto user;

    /**
     * The upload time of this post in ISO format e.g.2022-07-23T00:42:50Z .
     */
    private String uploadTime;

    private Long sharedPostId;

    private boolean notificationEnabled;

    private String tags;

    private Long totalComments;

    private Boolean isUpvoted;

    private Boolean isDownvoted;

    private Boolean isSaved;

    private boolean followed;
    private boolean anonymous;

    private boolean nsfw;

    private boolean moderating;

    private boolean followersOnly;
}

File: client/src/services/dtos/post-dto.ts
import { PostContentType } from '../../models/enums/post-content-type';
import SectionDto from './section-dto';
import { UserDto } from './user-dto';

export default interface PostDto {
  readonly id: number;

  readonly title?: string;

  readonly mediaUrl: string;

  readonly mediaType: string;

  readonly downvotes: number;

  readonly upvotes: number;

  readonly section: SectionDto;

  readonly nsfw: boolean;

  /**
   * The upload time of this post in ISO format e.g.2022-07-23T00:42:50Z .
   */
  readonly uploadTime: string;

  readonly tags: string;

  readonly text: string | null;

  readonly contentType: PostContentType;

  readonly isUpvoted: boolean;

  readonly isDownvoted: boolean;

  readonly totalComments: number;

  readonly followed: boolean;

  readonly user?: UserDto;

  readonly isSaved: boolean;

  readonly notificationEnabled: boolean;

  readonly anonymous: boolean;

  readonly sharedPostId: number | null;
}
